src/system_identity/iam_client.rs

```diff
@@ -0,0 +1,189 @@
+            http_client,
+            name,
+            organization_id,
+            system_identity_creation_uri,
+        }
+    }
+
+    fn system_identity_from_token(
+        http_client: &C,
+        name: &str,
+        organization_id: &str,
+        token: &str,    // type
+        pub_key: &[u8], // type and provenance, other modes accepted?
+        system_identity_creation_uri: &Uri,
+    ) -> Result<SystemIdentityCreationResponseData, HttpClientError> {
+        let pub_key_b64 = general_purpose::STANDARD.encode(pub_key);
+        let json_body_string = json!({
+            "query": format!(
+                "mutation {{ systemIdentityCreate(name: \"{}\", organizationId: \"{}\", publicKey: \"{}\") {{ clientId, name }} }}",
+                name, organization_id, pub_key_b64
+            ),
+        });
+        let json_body = serde_json::to_vec(&json_body_string)
+            .map_err(|e| HttpClientError::EncoderError(format!("Failed to encode JSON: {e}")))?;
+
+        let request = http::Request::builder()
+            .uri(system_identity_creation_uri)
+            .method("POST")
+            .header(CONTENT_TYPE, "application/json")
+            .header(AUTHORIZATION, format!("Bearer {token}"))
+            .body(json_body)
+            .map_err(|e| HttpClientError::EncoderError(format!("Failed to build request: {e}")))?;
+
+        let response = http_client.send(request)?;
+        let body = response.body();
+        match response.status() {
+            StatusCode::OK => {
+                let system_identity_response: SystemIdentityCreationResponseData =
+                    serde_json::from_slice(body).map_err(|e| {
+                        HttpClientError::DecoderError(format!("Failed to decode JSON: {e}"))
+                    })?;
+                Ok(system_identity_response)
+            }
+            status => Err(HttpClientError::UnsuccessfulResponse(
+                status.as_u16(),
+                String::from_utf8_lossy(body).to_string(),
+            )),
+        }
+    }
+}
+
+impl<C: HttpClient> IAMClient for HttpIAMClientImpl<'_, C> {
+    fn create_system_identity(
+        &self,
+        token: &AccessToken,
+        pub_key: &[u8],
+    ) -> Result<SystemIdentityCreationResponseData, IAMClientError> {
+        Self::system_identity_from_token(
+            self.http_client,
+            self.name.as_str(),
+            self.organization_id.as_str(),
+            token,
+            pub_key,
+            self.system_identity_creation_uri,
+        )
+        .map_err(|e| IAMClientError::IAMClientError(e.to_string()))
+    }
+}
+
+/// Represents the response data from the System Identity service
+#[derive(Debug, Clone, PartialEq)]
+pub struct SystemIdentityCreationResponseData {
+    pub client_id: String,
+    pub name: String,
+}
+
+/// Manual implementation of Deserialize to handle nested structure that comes as response
+/// from the System Identity service.
+// TODO do we need this data at all?
```

Comment:
The TODO seems to contradict the comment below:

>  (...)
>         So we create the appropriate intermediate structures to
>        deserialize it and return only the actual data we need.

Do we need the data or not? 🙃

Reply:
Removed in https://github.com/newrelic/newrelic-auth-rs/pull/25/commits/2b7a4d6fe472fa48f2ed2353fbb78636b47ded13